lib/http/errors.rb

```diff
@@ -18,6 +18,9 @@ class StateError < ResponseError; end
   # Generic Timeout error
   class TimeoutError < Error; end
 
-  # Header name is invalid
-  class InvalidHeaderNameError < Error; end
+  # Header value is of unexpected format (similar to Net::HTTPHeaderSyntaxError)
+  class HeaderError < Error; end
+
+  # @deprecated will be removed in v3.0.0
```

Comment:
As we're gonna release this PR as part of `v3.0.0` it worth to change this to be *will be removed in v3.1.0*